compiler/parser/RdrHsSyn.hs

```diff
@@ -3434,10 +3434,14 @@ mkInterfaceImplements templateName conName sharedBinds implements =
       } = implements
 
     implementsMarker =
-      let name =
+      let mangle = (>>= \x -> if (x == '_') then "__" else [x])
```

Comment:
as discussed offline, we're gonna need a similar fix for the `_method_` markers, so I'd suggest moving `mangle` to the top level, and making the `case` expression on `isQual_maybe` its own top level function too.